catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/builder/JsonModelBuilder.java

```diff
@@ -102,129 +82,271 @@ public FilterNode getResult() {
     return this;
   }
 
   @Override
-  public FlatFilterBuilder beginPropertyIsLikeType(String operator, boolean matchCase) {
-    verifyResultNotYetRetrieved();
-    verifyTerminalNodeNotInProgress();
-    if (!PROPERTY_IS_LIKE.equals(operator)) {
-      throw new IllegalArgumentException("Cannot find mapping for like operator: " + operator);
-    }
-    // For now, will always choose ILIKE
-    String jsonOperator = (matchCase) ? LIKE : ILIKE;
-    nodeInProgress = new FilterNodeImpl(jsonOperator);
+  public JsonModelBuilder isEqualTo(boolean matchCase) {
+    beginTerminalType("=");
     return this;
   }
 
   @Override
-  public FlatFilterBuilder beginBinaryTemporalType(String operator) {
-    verifyResultNotYetRetrieved();
-    verifyTerminalNodeNotInProgress();
-    String jsonOperator = BINARY_TEMPORAL_MAPPING.get(operator);
-    if (jsonOperator == null) {
-      throw new IllegalArgumentException(
-          "Cannot find mapping for binary temporal operator: " + operator);
-    }
-    nodeInProgress = new FilterNodeImpl(jsonOperator);
+  public JsonModelBuilder isNotEqualTo(boolean matchCase) {
+    beginTerminalType("!=");
     return this;
   }
 
   @Override
-  public JsonModelBuilder beginBinarySpatialType(String operator) {
+  public JsonModelBuilder isGreaterThan(boolean matchCase) {
+    beginTerminalType(">");
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder isGreaterThanOrEqualTo(boolean matchCase) {
+    beginTerminalType(">=");
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder isLessThan(boolean matchCase) {
+    beginTerminalType("<");
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder isLessThanOrEqualTo(boolean matchCase) {
+    beginTerminalType("<=");
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder like(
+      boolean matchCase, String wildcard, String singleChar, String escape) {
+    String operator = (matchCase) ? "LIKE" : "ILIKE";
+    beginTerminalType(operator);
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder before() {
+    beginTerminalType("BEFORE");
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder after() {
+    beginTerminalType("AFTER");
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder intersects() {
+    beginTerminalType("INTERSECTS");
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder dwithin(double distance, String units) {
+    throw new UnsupportedOperationException("Distance buffers in JSON are not yet supported");
+  }
+
+  @Override
+  public JsonModelBuilder function(String name) {
     verifyResultNotYetRetrieved();
-    verifyTerminalNodeNotInProgress();
-    if (!BINARY_SPATIAL_OPS.contains(operator)) {
-      throw new IllegalArgumentException("Invalid operator for binary spatial type: " + operator);
-    }
-    nodeInProgress = new FilterNodeImpl(operator);
+    verifyTerminalNodeInProgress(); // TODO - Verify this; per schema functions can be predicates
```

Comment:
❗️ Address TODO. Per version 2 of the Filter spec it appears the `Function` element is now a valid filter, both on its own and part of a logic operation. Either add the proper tests so this is supported functionality, or choose to not support it for the time being. 

_See companion comment in XmlModelBuilder as well_.

Reply:
Verified. Will be an accepted limitation since we do not need that level of expressivity.